### PR TITLE
feat(sonarqube): add file metrics convertor

### PR DESCRIPTION
**Pull request by warren830**
### Summary
add file metrics convertor

### Does this close any open issues?
part of #2305 

### Screenshots
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: backend/core/models/domainlayer/securitytesting/st_issues.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ type StIssue struct {
 	Rule              string           `json:"rule" gorm:"type:varchar(255)"`
 	Severity          string           `json:"severity" gorm:"type:varchar(255)"`
 	Component         string           `json:"component" gorm:"type:varchar(255)"`
-	Project           string           `json:"project" gorm:"type:varchar(255)"`
+	ProjectKey        string           `gorm:"index;type:varchar(255)"` //domain project key
```
Comment (abeizn) [inline backend/core/models/domainlayer/securitytesting/st_issues.go:32]: What's the differenct?

File: backend/plugins/sonarqube/tasks/hotspots_convertor.go [hunk 2]
```diff
@@ -47,19 +47,20 @@ func ConvertHotspots(taskCtx plugin.SubTaskContext) errors.Error {
...
 		Convert: func(inputRow interface{}) ([]interface{}, errors.Error) {
 			sonarqubeHotspot := inputRow.(*sonarqubeModels.SonarqubeHotspot)
 			domainHotspot := &securitytesting.StIssue{
-				DomainEntity:      domainlayer.DomainEntity{Id: accountIdGen.Generate(data.Options.ConnectionId, sonarqubeHotspot.Key)},
+				DomainEntity:      domainlayer.DomainEntity{Id: issueIdGen.Generate(data.Options.ConnectionId, sonarqubeHotspot.Key)},
 				BatchId:           sonarqubeHotspot.BatchId,
 				Key:               sonarqubeHotspot.Key,
 				Component:         sonarqubeHotspot.Component,
-				Project:           sonarqubeHotspot.Project,
+				ProjectKey:        projectIdGen.Generate(data.Options.ConnectionId, sonarqubeHotspot.Project),
```
Comment (abeizn) [inline backend/plugins/sonarqube/tasks/hotspots_convertor.go:63]: need connectionId?

File: backend/core/models/domainlayer/securitytesting/st_issues.go (hunk 1 above)
Comment (warren830) [reply backend/core/models/domainlayer/securitytesting/st_issues.go]: fixed

**Review by abeizn — APPROVED** [review]
LGTM

**Pull request merged**